internal/controller/clusterpolicy_controller.go

```diff
@@ -0,0 +1,179 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/edgedb/edgedb-go"
+	"github.com/pingcap/errors"
+
+	kyvernoV1 "github.com/kyverno/kyverno/api/kyverno/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/giantswarm/policy-meta-operator/internal/utils"
+	edgedbutils "github.com/giantswarm/policy-meta-operator/internal/utils/edgedb"
+)
+
+// PolicyReconciler reconciles a AutomatedException object
+type ClusterPolicyReconciler struct {
+	client.Client
+	EdgeDBClient     *edgedb.Client
+	Scheme           *runtime.Scheme
+	MaxJitterPercent int
+}
+
+var (
+	//GiantSwarm team label
+	GSTeamLabel = "application.giantswarm.io/team"
+	//Policy API Exemption label
+	//TODO: Move to Policy API
+	PolicyAPIExemptionLabel = "policy.giantswarm.io/giantswarm-exempt"
+)
+
+func (r *ClusterPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = log.FromContext(ctx)
+
+	var clusterPolicy kyvernoV1.ClusterPolicy
+
+	if err := r.Get(ctx, req.NamespacedName, &clusterPolicy); err != nil {
+		if !errors.IsNotFound(err) {
+			// Error fetching the report
+			log.Log.Error(err, "unable to fetch Kyverno ClusterPolicy")
+		} else {
+			// Policy not found, make sure we don't have it in edgedb either
+			err := edgedbutils.DeleteKyvernoClusterPolicy(ctx, r.EdgeDBClient, req.Name)
+			if err != nil {
+				log.Log.Error(err, "Error deleting ClusterPolicy from database")
+			}
+		}
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if clusterPolicy.Spec.ValidationFailureAction.Enforce() {
+		if clusterPolicy.HasValidate() {
+			log.Log.Info("Kyverno ClusterPolicy is in enforce mode and has validating rules, adding it to edgedb")
+
+			policyRuleNames := extractRuleNames(clusterPolicy)
+			policyTargetKinds := extractTargetKinds(clusterPolicy)
+			exempted := isExempted(clusterPolicy)
+
+			if len(policyRuleNames) == 0 || len(policyTargetKinds) == 0 {
+				log.Log.Error(errors.New("Error extracting rule names or target kinds"), fmt.Sprintf("Error extracting rule names or target kinds from Kyverno ClusterPolicy %s", clusterPolicy.Name))
+				return ctrl.Result{}, nil
+			}
+
+			_, err := edgedbutils.InsertKyvernoClusterPolicy(ctx, r.EdgeDBClient, clusterPolicy.Name, policyRuleNames, policyTargetKinds, exempted)
+			if err != nil {
+				log.Log.Error(err, "Error inserting Kyverno ClusterPolicy in database")
+			}
+
+		}
+	}
+
+	return utils.JitterRequeue(utils.DefaultRequeueDuration, r.MaxJitterPercent, log.Log), nil
+}
+
+func isExempted(clusterPolicy kyvernoV1.ClusterPolicy) bool {
```

Comment:
Nobody except us will understand what exempted means here :see_no_evil: 
```suggestion
func shouldExcludeGiantSwarmResources(clusterPolicy kyvernoV1.ClusterPolicy) bool {
```